mbuild/compound.py

```diff
@@ -932,6 +932,29 @@ def available_ports(self):
             if isinstance(p, Port) and not p.used
         ]
 
+    def direct_bonds(self):
+        """Return all bonds a particle is directly involved in.
+
+        Returns
+        -------
+        List of mb.Compound
+
+        See Also
+        --------
+        bond_graph.edges_iter : Iterations over all edges in a BondGraph
+        Compound.n_direct_bonds() : Returns the number of bonds a particle contains
+        """
+        if list(self.particles()) != [self]:
+            raise MBuildError(
+                "The direct_bonds method can only "
+                "be used on compounds at the bottom of their hierarchy."
+            )
+        if not self.root.bond_graph:
+            return iter(())
+        elif self.root.bond_graph.has_node(self):
+            for i in self.root.bond_graph._adj[self]:
+                yield i
```

Comment:
I think we are returning only the neighboring particle here but not the bond itself 
```suggestion
                yield (self, i)
```

Reply:
or if we want to keep this behavior (returning the neighboring particles), we can change the docstring for this method